torch/nn/modules/module.py

```diff
@@ -181,8 +183,45 @@ def register_module_backward_hook(
     return handle
 
 
+def register_module_backward_pre_hook(
+    hook: Callable[['Module', _grad_t], Union[None, _grad_t]]
+) -> RemovableHandle:
+    r"""Registers a backward pre-hook common to all the modules.
+
+    .. warning ::
+        This adds global state to the `nn.module` module
+        and it is only intended for debugging/profiling purposes.
+
+    The hook will be called every time before the gradients with respect
+    to module inputs are computed. The hook should have the following signature::
```

Comment:
Could you reword as: "The hook will be called every time the gradients for the module are computed."